waiter/src/waiter/state.clj

```diff
@@ -406,18 +406,37 @@
        :response-chan response-chan
        :response response-code})))
 
+(defn- unblacklist-instance
+  [{:keys [instance-id->state] :as current-state} update-instance-id->blacklist-expiry-time-fn update-status-tag-fn
+   instance-id expiry-time]
+  (log/info "unblacklisting instance" instance-id "as blacklist expired at" expiry-time)
+  (cond-> (update-instance-id->blacklist-expiry-time-fn current-state #(dissoc % instance-id))
+          (contains? instance-id->state instance-id)
+          (update-in [:instance-id->state instance-id] update-status-tag-fn #(disj % :blacklisted))))
+
 (defn handle-unblacklist-request
   "Handle a request to unblacklist an instance."
-  [{:keys [instance-id->blacklist-expiry-time instance-id->state] :as current-state} update-status-tag-fn 
+  [{:keys [instance-id->blacklist-expiry-time] :as current-state} update-status-tag-fn
    update-instance-id->blacklist-expiry-time-fn {:keys [instance-id]}]
-  (let [expiry-time (get instance-id->blacklist-expiry-time instance-id)
-        unblacklist? (and (not (nil? expiry-time)) (not (t/after? expiry-time (t/now))))]
-    (if unblacklist?
-      (do
-        (log/info "unblacklisting instance" instance-id "as blacklist expired at" expiry-time)
-        (cond-> (update-instance-id->blacklist-expiry-time-fn current-state #(dissoc % instance-id))
-          (contains? instance-id->state instance-id)
-          (update-in [:instance-id->state instance-id] update-status-tag-fn #(disj % :blacklisted))))
+  (let [expiry-time (get instance-id->blacklist-expiry-time instance-id)]
+    (if (utils/time-less-than-or-equal-to expiry-time (t/now))
+      (unblacklist-instance current-state update-instance-id->blacklist-expiry-time-fn update-status-tag-fn
+                            instance-id expiry-time)
+      current-state)))
+
+(defn handle-unblacklist-cleanup
```

Comment:
Rename to `cleanup-blacklist`.  `handle-*` to me implies it's a handler of a message of some kind.

Reply:
renamed to `handle-unblacklist-cleanup-request` to keep it consistent with the other helper methods that have the form `handle-*-request`